blog/content/post/2021-10-28-secure-jtag-on-imx-soc.md

```diff
@@ -0,0 +1,270 @@
+---
+title: 'Securing JTAG on i.MX6 and i.MX8MM'
+abstract: 'JTAG help a lot of engineers during product development.
+          It also may be helpful for hackers.
+          I tell you why and how to increase JTAG security in your product'
+cover: /covers/secure-jtag.png
+author: michal.kotyla
+layout: post
+published: true
+date: 2021-10-28
+archives: "2021"
+
+tags:
+  - security
+  - jtag
+  - fuse
+  - bit
+  - fusebit
+  - fusebits
+  - fusing
+  - fuses
+  - nxp
+  - imx
+  - imx6
+  - i.mx6
+  - imx8
+  - i.mx8
+  - imx8mm
+  - i.mx8mm
+categories:
+  - Firmware
+  - Miscellaneous
+  - OS Dev
+  - Security
+  - Manufacturing
+
+---
+## Introduction
+
+JTAG port is an important feature that help a lot of engineers during product
+development. It also may be helpful for hackers while reverse-engineering
+process - e.g. they can dump the memory of your firmware and get access to
+strictly confidential information. In this article, I show you how to lock JTAG
+access for users who don't have a special key (password). I use for that Secure
+JTAG feature which is implemented on i.MX SoC's. In our lab, we tested this
+solution on i.MX6 and i.MX8MM. Probably this same feature is available on i.MX8
+but unfortunately security reference manual for this SoC are not publicly
+available.
+
+## Requirements
+
+- **Development board**: Any board or devkit with i.MX6 and i.MX8MM SoC. For
+testing this solution or i.MX6 we use our internal, custom board. For Secure
+JTAG on i.MX8M we use `DART-MX8M-MINI` from
+[Variscite](https://variwiki.com/index.php?title=DART-MX8M-MINI),
+
+- **SoC reference manual**: Setting JTAG in secure mode is a dangerous operation
+please check twice that everything is correct. Read SoC documentation and
+make sure that you will not brick the device,
+
+- **crucible tool**: JTAG mode and access key are saved in fuse bits - we need
+software to manipulate this in userspace. I used
+[crucible](https://github.com/f-secure-foundry/crucible/) for that, but I show
+you how to change one-time-programmable fields from U-boot,
```

Comment:
Let's show how to do it from U-Boot and crucible

Reply:
ok